specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerAppsPatches.json

```diff
@@ -0,0 +1,566 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "ContainerAppsPatches_Apply_0": {
+            "$ref": "./examples/ContainerAppsPatches_Apply.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ContainerAppsPatchResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PatchProperties",
+          "description": "Properties that describes current states of the patch resource.",
+          "x-ms-mutability": [
+            "read",
+            "update"
+          ]
+        }
+      },
+      "description": "Container App Patch",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "PatchProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled"
+      ],
+      "x-ms-enum": {
+        "name": "PatchProvisioningState",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchApplyStatus": {
+      "type": "string",
+      "description": "The status of the patch once it has been provisioned",
+      "enum": [
+        "NotStarted",
+        "RebaseInProgress",
+        "CreatingRevision",
+        "Succeeded",
+        "Canceled",
+        "RebaseFailed",
+        "RevisionCreationFailed",
+        "ImagePushPullFailed",
+        "ManuallySkipped"
+      ],
+      "x-ms-enum": {
+        "name": "PatchApplyStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "DetectionStatus": {
+      "type": "string",
+      "description": "The status of the patch detection.",
+      "enum": [
+        "Succeeded",
+        "RegistryLoginFailed",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "DetectionStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchDetails": {
+      "description": "The detailed info of patch operation performing when applying a patch.",
+      "type": "object",
+      "required": [
+        "targetContainerName",
+        "targetImage",
+        "lastDetectionTime",
+        "detectionStatus"
+      ],
+      "properties": {
+        "targetContainerName": {
+          "type": "string",
+          "description": "The name of the target container for the patch.",
+          "readOnly": true
+        },
+        "targetImage": {
+          "type": "string",
+          "description": "The name of the target image for the patch.",
+          "readOnly": true
+        },
```

Comment:
I am guessing these are references to azure resources? Consider instead using full azure resource ids, so that its a flexible API with respect to resource container, like subscription, resource group, management group.